esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -38,10 +38,73 @@
 #include "build-config.h"
 #include "jsmartsock.ipp"
 #include "esdl_monitor.hpp"
+#include "EsdlAccessMapGenerator.hpp"
 
 #include "loggingagentbase.hpp"
 #include "httpclient.ipp"
 
+class EsdlSvcReporter : public EsdlDefReporter
+{
+protected:
+    void reportSelf(Flags flag, const char* component, const char* level, const char* msg) const override
+    {
+        const LogMsgCategory* catPtr = nullptr;
+        switch (flag & ReportCategoryMask)
+        {
+        case ReportDisaster:  catPtr = &MCdisaster; break;
+        case ReportAError:    catPtr = &MCauditError; break;
+        case ReportIError:    catPtr = &MCinternalError; break;
+        case ReportOError:    catPtr = &MCoperatorError; break;
+        case ReportUError:    catPtr = &MCuserError; break;
+        case ReportIWarning:  catPtr = &MCinternalWarning; break;
+        case ReportOWarning:  catPtr = &MCoperatorWarning; break;
+        case ReportUWarning:  catPtr = &MCuserWarning; break;
+        case ReportDProgress: catPtr = &MCdebugProgress; break;
+        case ReportOProgress: catPtr = &MCoperatorProgress; break;
+        case ReportUProgress: catPtr = &MCuserProgress; break;
+        case ReportDInfo:     catPtr = &MCdebugInfo; break;
+        case ReportOInfo:     catPtr = &MCoperatorInfo; break;
+        case ReportUInfo:     catPtr = &MCuserInfo; break;
+        case ReportStats:     catPtr = &MCstats; break;
+        default: return;
+        }
+        LOG(*catPtr, "%s [%s]: %s", level, component, msg);
+    }
+};
+
+static const char* Scope_EsdlService = "EsdlService";
+static const char* Scope_EsdlMethod = "EsdlMethod";
+static const char* Scope_BindingService = "BindingService";
+static const char* Scope_BindingMethod = "BindingMethod";
+static const char* Var_Service = "service";
+static const char* Var_Method = "method";
+#define ANNOTATION_AUTH_FEATURE "auth_feature"
+
+// The scope mapper and level mapper instances could be static service members, but enabling the
+// declaration of these types in the header requires inclusion of headers that unnecessarily affect
+// other projects that include this class' header.
```

Comment:
What effect do you see on other projects when they're static service members?

Reply:
To make these instances static members of the service class would require the header included on line 41 to be included from the service header file. That change requires all projects that reference the header to change the CMakeLists file to include the hidl directory to get the file included by the include. Otherwise those projects won't compile.

I think static members might appear to be a bit "cleaner", but I decided the apparent benefit was outweighed by the cost to change unrelated projects. I added this and the following comment to address that and to preempt questions about static versus instance/local usage.